src/sinks/aws_kinesis_streams/request_builder.rs

```diff
@@ -2,31 +2,35 @@ use std::io;
 
 use aws_sdk_kinesis::{model::PutRecordsRequestEntry, types::Blob};
 use bytes::Bytes;
+use vector_common::request_metadata::{MetaDescriptive, RequestMetadata};
 use vector_core::ByteSizeOf;
 
-use super::sink::KinesisProcessedEvent;
+use super::sink::{KinesisKey, KinesisProcessedEvent};
 use crate::{
     codecs::{Encoder, Transformer},
     event::{Event, EventFinalizers, Finalizable},
-    sinks::util::{request_builder::EncodeResult, Compression, RequestBuilder},
+    sinks::util::{
+        metadata::RequestMetadataBuilder, request_builder::EncodeResult, Compression,
+        RequestBuilder,
+    },
 };
 
 pub struct KinesisRequestBuilder {
     pub compression: Compression,
     pub encoder: (Transformer, Encoder<()>),
 }
 
-pub struct Metadata {
+pub struct KinesisMetadata {
     pub finalizers: EventFinalizers,
     pub partition_key: String,
-    pub event_byte_size: usize,
 }
 
 #[derive(Clone)]
 pub struct KinesisRequest {
+    pub key: KinesisKey,
     pub put_records_request: PutRecordsRequestEntry,
     pub finalizers: EventFinalizers,
-    pub event_byte_size: usize,
+    metadata: RequestMetadata,
```

Comment:
Does every request struct contain this metadata? Could it be carried around by the framework (ie a wrapper struct containing the request + metadata) rather than the individual sinks?